pkg/plan/recipe/upgrade_plans.go

```diff
@@ -0,0 +1,113 @@
+			"upgrade:node-unlock-kubernetes",
+			&resource.Run{Script: object.String("apt-mark unhold 'kube*' || true")})
+		b.AddResource(
+			"upgrade:node-install-kubeadm",
+			&resource.Deb{Name: "kubeadm", Suffix: "=" + k8sVersion + "-00"},
+			plan.DependOn("upgrade:node-unlock-kubernetes"))
+	}
+	//
+	// For secondary masters
+	// version >= 1.16.0 uses: kubeadm upgrade node
+	// version >= 1.14.0 && < 1.16.0 uses: kubeadm upgrade node experimental-control-plane
+	//
+	secondaryMasterUpgradeControlPlaneFlag := ""
+	if lt, err := version.LessThan(k8sVersion, "v1.16.0"); err == nil && lt {
+		secondaryMasterUpgradeControlPlaneFlag = "experimental-control-plane"
+	}
+
+	switch ntype {
+	case OriginalMaster:
+		b.AddResource(
+			"upgrade:node-kubeadm-upgrade",
+			&resource.Run{Script: object.String(fmt.Sprintf("kubeadm upgrade plan && kubeadm upgrade apply -y %s", k8sVersion))},
+			plan.DependOn("upgrade:node-install-kubeadm"))
+	case SecondaryMaster:
+		b.AddResource(
+			"upgrade:node-kubeadm-upgrade",
+			&resource.Run{Script: object.String(fmt.Sprintf("kubeadm upgrade node %s", secondaryMasterUpgradeControlPlaneFlag))},
+			plan.DependOn("upgrade:node-install-kubeadm"))
+	case Worker:
+		b.AddResource(
+			"upgrade:node-kubeadm-upgrade",
+			&resource.Run{Script: object.String(fmt.Sprintf("kubeadm upgrade node config --kubelet-version %s", k8sVersion))},
+			plan.DependOn("upgrade:node-install-kubeadm"))
+	}
+
+	switch pkgType {
+	case resource.PkgTypeRPM, resource.PkgTypeRHEL:
+		b.AddResource(
+			"upgrade:node-kubelet",
+			&resource.RPM{Name: "kubelet", Version: k8sVersion, DisableExcludes: "kubernetes"},
+			plan.DependOn("upgrade:node-kubeadm-upgrade"))
+		b.AddResource(
+			"upgrade:node-restart-kubelet",
+			&resource.Run{Script: object.String("systemctl restart kubelet")},
+			plan.DependOn("upgrade:node-kubelet"))
+		b.AddResource(
+			"upgrade:node-kubectl",
+			&resource.RPM{Name: "kubelet", Version: k8sVersion, DisableExcludes: "kubernetes"},
+			plan.DependOn("upgrade:node-restart-kubelet"))
+		b.AddResource(
+			"upgrade:node-lock-kubernetes",
+			&resource.Run{Script: object.String("yum versionlock add 'kube*' || true")},
+			plan.DependOn("upgrade:node-kubectl"))
+	case resource.PkgTypeDeb:
+		b.AddResource(
+			"upgrade:node-kubelet",
+			&resource.Deb{Name: "kubelet", Suffix: "=" + k8sVersion + "-00"},
+			plan.DependOn("upgrade:node-kubeadm-upgrade"))
+		b.AddResource(
+			"upgrade:node-restart-kubelet",
+			&resource.Run{Script: object.String("systemctl restart kubelet")},
+			plan.DependOn("upgrade:node-kubelet"))
+		b.AddResource(
+			"upgrade:node-kubectl",
+			&resource.Deb{Name: "kubectl", Suffix: "=" + k8sVersion + "-00"},
+			plan.DependOn("upgrade:node-restart-kubelet"))
+		b.AddResource(
+			"upgrade:node-lock-kubernetes",
+			&resource.Run{Script: object.String("apt-mark hold 'kube*' || true")},
+			plan.DependOn("upgrade:node-kubectl"))
+	}
+
+	p, err := b.Plan()
+	if err != nil {
+		log.Fatalf("%v", err)
```

Comment:
I think this should return an error. We use `log.Fatalf` in the initial master plan creation because any failure there mean we can't install the cluster so there's no reason to continue. In this case, however, we probably don't want to kill the controller...